tools/sotn_build/get_file_list.py

```diff
@@ -0,0 +1,311 @@
+#!/usr/bin/env python3
+
+import argparse
+import yaml
+import sys
+import os
+import re
+from string import Template
+
+# I intend things like this to be stored in a conf file when this functionality is more established in the project
+# Since it has a high likelihood of changing and isn't very complex, keeping it here is easier for now
+CONFIG_BASE_DIR: str = "config"
+CONFIG_PATTERNS: dict[Template] = {
+    "splat": Template("splat.$version.(?:st|bo)?$ovl.yaml")
+}
+# I would prefer os.path.join(path,filename.ext) because it is the safer method, but it
+# further muddies set comprehensions that are already difficult to read in some cases
+FILENAME_TEMPLATE: Template = Template("$path/$filename.$ext")
+SEG_TYPES: dict[tuple] = {
+    "asm": ("asm", "header"),
+    "data": ("data", "rodata", "bss", "sbss"),
+    "src": ("c"),
+    "asset": (
+        "cmp",
+        "ci4",
+        "palette",
+        "raw",
+        "assets",
+        "bin",
+    ),
+}
+
+
+# This cannot handle returning only shared src files, but since there aren't any currently it isn't really worth putting time into it.
```

Comment:
I'm not clear on this comment, is this some sort of antiquity that is no longer used in the codebase but was present in the Makefile so it's made its way here too?

From my understanding all the shared src files are coming via .h imports from overlay specific .c files?

Reply:
You are correct on both accounts.  I had already written handling of shared source files before I realized there were none.